### PR TITLE
docs: update event api docs 

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:  #21146

## What is the new behavior?
Doc added 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jenniferfell** [comment]
@IgorMinar @kara Hi. As part of her work on API templates and examples, @jbogarthyde took a pass at rewriting the event API docs. This work depends on API markup, generator, and output work here in master, but the content updates are not specific to v6. Looking for technical review...accurate? complete? makes sense? Hoping to use this as an example for API rewriting ed for team, and bonus...we can release these updates to the world (after review-update cycle is complete). Thanks.

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 1]
```diff
@@ -11,34 +11,54 @@ import {Inject, Injectable, InjectionToken, NgZone} from '@angular/core';
 import {getDOM} from '../dom_adapter';
 
 /**
+ * The injection token for the event-manager plug-in service.
```
Comment (vicb) [inline packages/platform-browser/src/dom/events/event_manager.ts]: Should be "The injection token for the event-manager plug-in*s*" 
(ie not a service but multiple plugins)

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 2]
```diff
@@ -11,34 +11,54 @@ import {Inject, Injectable, InjectionToken, NgZone} from '@angular/core';
...
  * @stable
  */
 export const EVENT_MANAGER_PLUGINS =
     new InjectionToken<EventManagerPlugin[]>('EventManagerPlugins');
 
 /**
+ * @description An injectable service that provides event management in the browser.
```
Comment (vicb) [inline packages/platform-browser/src/dom/events/event_manager.ts]: `A service that delegates even management to {@link EventManagerPlugin}s`

File: packages/platform-browser/src/dom/events/event_manager.ts (hunk 1 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/event_manager.ts]: done

File: packages/platform-browser/src/dom/events/event_manager.ts (hunk 2 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/event_manager.ts]: done (with typo corrected)

File: packages/platform-browser/src/dom/events/event_manager.ts (hunk 1 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/event_manager.ts]: Done, thanks.

File: packages/platform-browser/src/dom/events/event_manager.ts (hunk 2 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/event_manager.ts]: Done, thanks.

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 3]
```diff
@@ -95,15 +134,33 @@ export class HammerGestureConfig {
...
+ * @description An injectable service that supports {@link http://hammerjs.github.io/ HammerJS}
+ * gesture recognition in Angular.
+ */
 @Injectable()
 export class HammerGesturesPlugin extends EventManagerPlugin {
+  /**
+   * Initializes an instance of the Hammer gestures-manager service.
+   * @param doc The document in which to detect Hammer gestures.
+   * @param _config The supported-gestures configuration.
+   * @param console The logging console.
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: any constructor of an `@Injectable` class is a private api and should not be documented.

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 4]
```diff
@@ -11,34 +11,54 @@ import {Inject, Injectable, InjectionToken, NgZone} from '@angular/core';
...
  */
 @Injectable()
 export class EventManager {
   private _plugins: EventManagerPlugin[];
   private _eventNameToPlugin = new Map<string, EventManagerPlugin>();
-
+  /**
+   * Initializes an instance of the event-manager service.
+   * @param plugins The event-management plug-ins to provide.
+   * @param _zone The compilation zone.
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/event_manager.ts]: please don't document constructors of injectable classes - the constructor parameters of injectable classes are an implementation detail and not a public api.

File: packages/platform-browser/src/dom/events/key_events.ts [hunk 5]
```diff
@@ -23,13 +32,31 @@ const MODIFIER_KEY_GETTERS: {[key: string]: (event: KeyboardEvent) => boolean} =
 
 /**
  * @experimental
+ * @description An injectable service that provides support for handling of key events in Angular.
  */
 @Injectable()
 export class KeyEventsPlugin extends EventManagerPlugin {
+  /**
+   * Initializes an instance of the key-event management service.
+   * @param doc The document in which key events will be detected.
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/key_events.ts]: no docs

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 6]
```diff
@@ -95,15 +134,33 @@ export class HammerGestureConfig {
   }
 }
 
+// This doesn't show up in generated doc
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: is this a public api?

**Review by IgorMinar — CHANGES_REQUESTED** [review]
please update constructor docs

**Comment by IgorMinar** [comment]
hi there, this PR needs rebasing and previous review comments have not yet been addressed.

**Comment by jbogarthyde** [comment]
This is too old to sync - closing and creating a new PR

**Pull request closed without merging**